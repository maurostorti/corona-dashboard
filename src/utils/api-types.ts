export interface ApiData {
  confirmed: { value: any }
  recovered: { value: any }
  deaths: { value: any }
  lastUpdate: string
}

export interface CountryType {
  name: string
  iso2?: string
  iso3?: string
}

export interface DailyDataType {
  confirmed: {
    total: number
  }
  deaths: {
    total: number
  }
  recovered: {
    total: number
  }
  reportDate: string
}
