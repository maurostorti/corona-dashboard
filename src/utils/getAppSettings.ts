export default function getAppSettings() {
  return {
    apiURL: 'https://covid19.mathdro.id/',
  }
}
