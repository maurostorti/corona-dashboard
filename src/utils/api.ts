import axios from 'axios'
import getAppSettings from './getAppSettings'

export const { apiURL } = getAppSettings()

export const api = axios.create({
  baseURL: apiURL,
  // timeout: 0,
})

api.interceptors.request.use(
  (request) => {
    return request
  },
  (error) => {
    // Do something with response error
    console.error('ERROR RECEIVED', error.response)
    return Promise.reject(error)
  },
)
