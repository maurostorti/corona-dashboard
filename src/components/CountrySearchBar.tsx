import { Autocomplete, AutocompleteChangeReason, TextField } from '@mui/material'
import { useEffect, useState } from 'react'
import { api } from '../utils/api'
import '../App.css'
import { CountryType } from '../utils/api-types'

type CountrySearchBarProps = {
  onCountryChange: (country: CountryType | null, reason: AutocompleteChangeReason) => void
}
export default function CountrySearchBar({ onCountryChange }: CountrySearchBarProps) {
  const [countries, setCountries] = useState<CountryType[]>([])

  const getCountries = () => {
    api
      .get('api/countries')
      .then((response) => {
        const { countries: countriesData }: { countries: CountryType[] } = response.data
        setCountries(countriesData)
      })
      .catch((error) => console.error(`Error: ${error}`))
  }
  useEffect(() => {
    getCountries()
  }, [])

  return (
    <div className='Data-search'>
      <Autocomplete
        options={countries}
        getOptionLabel={(option) => option.name}
        autoComplete
        autoHighlight
        fullWidth
        renderInput={(params) => <TextField {...params} label='Search Countries' />}
        onChange={(event, value, reason) => onCountryChange(value, reason)}
      />
    </div>
  )
}
