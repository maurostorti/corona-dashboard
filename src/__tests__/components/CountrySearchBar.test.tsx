import { render } from '@testing-library/react'
import { AutocompleteChangeReason } from '@mui/material/Autocomplete'
import nock from 'nock'
import axios from 'axios'
import CountrySearchBar from '../../components/CountrySearchBar'
import { CountryType } from '../../utils/api-types'

// axios.defaults.adapter = require('axios/lib/adapters/http')

const onCountryChangeTest = (value: CountryType | null, reason: AutocompleteChangeReason) => null
const countries = {
  countries: [{ name: 'Afghanistan', iso2: 'AF', iso3: 'AFG' }, { name: 'West Bank and Gaza' }],
}

test('it renders correctly', () => {
  const { asFragment } = render(<CountrySearchBar onCountryChange={onCountryChangeTest} />)
  expect(asFragment()).toMatchSnapshot()
})

// Cannot Get nock to work

test.skip('it fetches data', async () => {
  const scope = nock('https://covid19.mathdro.id').get('/api/countries').reply(200, countries)

  await render(<CountrySearchBar onCountryChange={onCountryChangeTest} />)
  scope.done()
})
