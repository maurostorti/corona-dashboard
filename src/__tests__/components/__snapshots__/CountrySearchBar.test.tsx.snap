// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Data-search"
  >
    <div
      class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-eozd4h-MuiAutocomplete-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Search Countries
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="both"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Search Countries
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it renders correctly confirmed 1`] = `
<DocumentFragment>
  <div
    class="Data-search"
  >
    <div
      class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-eozd4h-MuiAutocomplete-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Search Countries
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="both"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Search Countries
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
